### PR TITLE
refactor: lock dependencies and cleanup code

### PR DESCRIPTION
**Pull request by ruisaraiva19**
Things updated:
- Favicons, they now have a white background to fix dark mode contrast issues
- Improved project code consistency by formatting all files via ESLint + Prettier
- All dependencies versions on `package.json` are now locked for consistency and to prevent future packages upgrade issues
- Automatically lint/format code on the `pre-commit` git hook
- PWA `theme_color` updated and added `maskable` property to icons

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  "editor.formatOnSave": true,
+  "eslint.validate": ["javascript", "javascriptreact"],
+  "eslint.run": "onType",
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  },
+  "css.validate": false,
+  "less.validate": false,
+  "scss.validate": false
+}
```
Comment (sexta13) [inline .vscode/settings.json:11]: this is optional by user, right? should not be commited.

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "recommendations": [
+    "mikestead.dotenv",
+    "dbaeumer.vscode-eslint",
+    "esbenp.prettier-vscode",
+    "jpoissonnier.vscode-styled-components"
+  ]
+}
```
Comment (sexta13) [inline .vscode/extensions.json:8]: same as below. User preferences....maybe put it in gitignore ;)

File: .vscode/settings.json (hunk 1 above)
Comment (ruisaraiva19) [reply .vscode/settings.json]: Deleted!

File: .vscode/extensions.json (hunk 2 above)
Comment (ruisaraiva19) [reply .vscode/extensions.json]: Deleted!

**Pull request merged**